app/src/main/java/org/oppia/app/options/StoryTextSizeActivityPresenter.kt

```diff
@@ -18,98 +14,21 @@ class StoryTextSizeActivityPresenter @Inject constructor(
-      )
-
-    fontSize = prefSummaryValue
-
-    binding.storyTextSizeToolbar.setNavigationOnClickListener {
-      val intent = Intent()
-      intent.putExtra(KEY_MESSAGE_STORY_TEXT_SIZE, fontSize)
-      (activity as StoryTextSizeActivity).setResult(REQUEST_CODE_TEXT_SIZE, intent)
-      activity.finish()
-    }
-
-    when (prefSummaryValue) {
-      getStoryTextSize(StoryTextSize.SMALL_TEXT_SIZE) -> {
-        binding.storyTextSizeSeekBar.progress = 0
-        binding.previewTextview.textSize = getStoryTextSizeInFloat(StoryTextSize.SMALL_TEXT_SIZE)
-      }
-      getStoryTextSize(StoryTextSize.MEDIUM_TEXT_SIZE) -> {
-        binding.storyTextSizeSeekBar.progress = 5
-        binding.previewTextview.textSize = getStoryTextSizeInFloat(StoryTextSize.MEDIUM_TEXT_SIZE)
-      }
-      getStoryTextSize(StoryTextSize.LARGE_TEXT_SIZE) -> {
-        binding.storyTextSizeSeekBar.progress = 10
-        binding.previewTextview.textSize = getStoryTextSizeInFloat(StoryTextSize.LARGE_TEXT_SIZE)
-      }
-      getStoryTextSize(StoryTextSize.EXTRA_LARGE_TEXT_SIZE) -> {
-        binding.storyTextSizeSeekBar.progress = 15
-        binding.previewTextview.textSize =
-          getStoryTextSizeInFloat(StoryTextSize.EXTRA_LARGE_TEXT_SIZE)
-      }
-    }
-
-    binding.storyTextSizeSeekBar.max = 15
-
-    binding.storyTextSizeSeekBar.setOnSeekBarChangeListener(
-      object :
-        SeekBar.OnSeekBarChangeListener {
-        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
-          var progressValue = progress
-          progressValue /= 5
-          progressValue *= 5
-
-          when (progressValue) {
-            0 -> {
-              fontSize = getStoryTextSize(StoryTextSize.SMALL_TEXT_SIZE)
-              binding.previewTextview.textSize =
-                getStoryTextSizeInFloat(StoryTextSize.SMALL_TEXT_SIZE)
-            }
-            5 -> {
-              fontSize = getStoryTextSize(StoryTextSize.MEDIUM_TEXT_SIZE)
-              binding.previewTextview.textSize =
-                getStoryTextSizeInFloat(StoryTextSize.MEDIUM_TEXT_SIZE)
-            }
-            10 -> {
-              fontSize = getStoryTextSize(StoryTextSize.LARGE_TEXT_SIZE)
-              binding.previewTextview.textSize =
-                getStoryTextSizeInFloat(StoryTextSize.LARGE_TEXT_SIZE)
-            }
-            else -> {
-              fontSize = getStoryTextSize(StoryTextSize.EXTRA_LARGE_TEXT_SIZE)
-              binding.previewTextview.textSize =
-                getStoryTextSizeInFloat(StoryTextSize.EXTRA_LARGE_TEXT_SIZE)
-            }
-          }
-          seekBar.progress = progressValue
-        }
-
-        override fun onStartTrackingTouch(seekBar: SeekBar) {
-        }
-
-        override fun onStopTrackingTouch(seekBar: SeekBar) {
-        }
-      })
+    activity.setContentView(R.layout.story_text_size_activity)
+    val storyTextSizeFragment = StoryTextSizeFragment.newInstance(prefSummaryValue)
```

Comment:
Please check if the fragment already exists or not, similar to `StoryActivityPresenter`

Reply:
This should be implemented in your other two PRs too.

Reply:
Done